client/js/actions/user-snippets-actions.js

```diff
@@ -5,15 +5,17 @@ import FlashMessages from './flash-messages-actions';
 class UserSnippetsActions {
   constructor(){}
 
-  getAllOfCurrentUser() {
-    axios.get('/snippets/user')
-      .then(res => {
-        FlashMessages.pushMessage({ content: 'Here are your own snippets!' });
-        this.dispatch({ok: true, snippets: res.data});
-      }).catch(err => {
-        FlashMessages.pushMessage({ content: 'Oops! Something went wrong :(' });
-        this.dispatch({ok: false, error: err});
-      });
+  getPaginatedUserSnippets(page, perPage){
```

Comment:
@dwoznicka you are repeating yourself consider moving that offset counting and url bulding to some util pagination lib and importing it and using it.